benchmarks/src/bitset_to_string.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <array>
+#include <benchmark/benchmark.h>
+#include <bit>
+#include <bitset>
+#include <random>
+
+using namespace std;
+
+namespace {
+    const auto random_bits = [] {
+        mt19937_64 rnd{};
+        array<uint64_t, 32> data;
+        for (auto& d : data) {
+            d = rnd();
+        }
+        return data;
+    }();
+
+    template <size_t N, class charT>
+    void BM_bitset_to_string(benchmark::State& state) {
+        for (auto _ : state) {
+            for (auto bits : random_bits) {
+                bitset<N> bs{bits};
+                benchmark::DoNotOptimize(bs.to_string<charT>());
+            }
+        }
+    }
+
+    template <class charT>
+    void BM_bitset_to_string_large_single(benchmark::State& state) {
+        const auto large_bitset = bit_cast<bitset<CHAR_BIT * sizeof random_bits>>(random_bits);

```

Comment:
No change requested: Technically, I don't believe that this is Standard. \[bit.cast\] is constrained for trivially copyable types, but \[template.bitset\] doesn't guarantee that for `bitset`. That said, `bit_cast` would refuse to compile otherwise, so I think we can get away with it.